### PR TITLE
Remove version file and switch to importlib.metadata

### PR DESCRIPTION
**Pull request by ChihweiLHBird**
1. Remove version file and switch to importlib.metadata.
2. Make Python 3.5 outdated.

### CODE REVIEW

File: sciunit/__init__.py [hunk 1]
```diff
@@ -13,7 +13,6 @@
 from .errors import Error
 from .scores.collections import ScoreArray, ScoreMatrix#, ScorePanel
 from .scores.collections_m2m import ScoreArrayM2M, ScoreMatrixM2M
-from .version import __version__
```
Comment (rgerkin) [inline sciunit/__init__.py:16]: Doing `import sciunit; sciunit.__version__` is something I would still like to support.  I think checking the `__version__` attribute of a top-level module is a pretty standard way that people find out the version number.  So can you add:
```
from importlib.metadata import version
__version__ = version('sciunit')
```
here?

**Review by rgerkin — CHANGES_REQUESTED** [review]
One change.

**Comment by rgerkin** [comment]
@ChihweiLHBird Also in addition the one above, can you remove 3.5 from [here](https://github.com/scidash/sciunit/blob/master/.github/workflows/python-package.yml#L18).

File: sciunit/__init__.py (hunk 1 above)
Comment (ChihweiLHBird) [reply sciunit/__init__.py:16]: Added.

**Comment by ChihweiLHBird** [comment]
Thank you, and all done. Can you help me to re-review it?

**Pull request merged**